prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -530,44 +558,96 @@ class HttpClientUtils {
+         * @param bearerTokens the access token needed to call the endpoint
          * @return a HttpClient with all sensible defaults
          */
-        fun createDefaultHttpClient(accessToken: String?): HttpClient {
-            return HttpClient(Apache) {
-                // installs logging into the call to post to the server
-                // commented out - not to override underlying default logger settings
-                // enable to trace http client internals when needed
-                // install(Logging) {
-                //     logger = Logger.SIMPLE
-                //     level = LogLevel.INFO
-                // }
-                // not using Bearer Auth handler due to refresh token behavior
-                accessToken?.let {
-                    defaultRequest {
-                        header("Authorization", "Bearer $it")
+        fun getDefaultHttpClient(accessToken: String?): HttpClient {
+            synchronized(this) {
+                if (accessToken != null) {
+                    return getDefaultHttpClientWithAuth(accessToken)
+                } else {
+                    httpClient ?: HttpClient(Apache) {
+                        install(ContentNegotiation) {
+                                json(
+                                    Json {
+                                        prettyPrint = true
+                                        isLenient = true
+                                        ignoreUnknownKeys = true
+                                    }
+                                )
+                            }
+                        install(HttpTimeout)
+                        engine {
+                                followRedirects = true
+                                socketTimeout = TIMEOUT
+                                connectTimeout = TIMEOUT
+                                connectionRequestTimeout = TIMEOUT
+                                customizeClient {
+                                }
+                            }
+                    }.also {
+                        httpClient = it
                     }
+                    return httpClient!!
                 }
-                install(ContentNegotiation) {
-                    json(
-                        Json {
-                            prettyPrint = true
-                            isLenient = true
-                            ignoreUnknownKeys = true
-                        }
-                    )
-                }
+            }
+        }
 
-                install(HttpTimeout)
-                engine {
-                    followRedirects = true
-                    socketTimeout = TIMEOUT
-                    connectTimeout = TIMEOUT
-                    connectionRequestTimeout = TIMEOUT
-                    customizeClient {
+        /**
+         * Called by getDefaultHttpClient as a helper to handle clients with auth tokens. Caller handles thread safety.
```

Comment:
Minor point of confusion... here it says "Caller handles thread safety" but later seems to say that safety is enforced by the way this class is written?

Reply:
I can rework the language. The intention of that comment is two fold: 

1. to alert future-David and other devs that the helper method doesn't enforce thread safety on client obj creation, the caller of the helper does by way of the `synchronized` block. 
2. to remind future-David and other devs that java/kotlin work in a way that prevents obj arguments passed to a method from changing unexpectedly by subsequent calls that change the source of that obj argument and thus THIS race-condition concern is not a concern. 

in other words I'm attempting to document the two likely sources of race-condition shennanigans: 

(1) speaks to what happens when two threads call the "get" method at the same time and why there's not going to be a race condition resulting in a caller receiving an object with arguments baked into it that it doesn't expect to be there. 

(2) speaks to what happens when a caller receives a client obj from the "get" method and, before it can use it to make a network call, a second caller receives a client obj with different arguments from the "get" method. 

I'll update the comment block to see if I can't make it more lucid.